src/v2/System/useFeatureFlag.tsx

```diff
@@ -60,15 +60,18 @@ export function useTrackVariantView({
   contextOwnerSlug,
   shouldTrackExperiment = true,
 }: VariantTrackingProperties) {
-  const { trackEvent } = useTracking()
-
   useEffect(() => {
-    if (typeof window !== "undefined") {
-      const trackFeatureView = shouldTrack(experimentName, variantName)
-
+    if (
+      typeof window !== "undefined" &&
+      typeof window.analytics !== "undefined"
+    ) {
+      const trackFeatureView =
+        shouldTrackExperiment && shouldTrack(experimentName, variantName)
```

Comment:
Added this extra check here to prevent saving the experiment in localstorage when the user is in a page that is not showing the experiment (eg. in our case we want the experiment to be displayed in the artist page although without this check if a user visited a page with an artwork grid the tracking event wasnt dispatched but we saved the experiment in the local storage which caused the inability to dispatch the tracking event when the user viewed the page with the experiment)